packages/admin/cms-admin/src/dam/FileForm/ReplaceFileButton.tsx

```diff
@@ -0,0 +1,98 @@
+import { useApolloClient } from "@apollo/client";
+import { useErrorDialog } from "@comet/admin";
+import { ThreeDotSaving, Upload } from "@comet/admin-icons";
+import { Button } from "@mui/material";
+import axios, { CancelTokenSource } from "axios";
+import { useRef, useState } from "react";
+import { FileRejection, useDropzone } from "react-dropzone";
+import { FormattedMessage } from "react-intl";
+
+import { useCmsBlockContext } from "../../blocks/useCmsBlockContext";
+import { replaceById } from "../../form/file/upload";
+import { convertMimetypesToDropzoneAccept } from "../DataGrid/fileUpload/fileUpload.utils";
+import { DamFileDetails } from "./EditFile";
+
+interface ReplaceFileButtonProps {
+    file: DamFileDetails;
+}
+
+export function ReplaceFileButton({ file }: ReplaceFileButtonProps) {
+    const apolloClient = useApolloClient();
+    const cmsBlockContext = useCmsBlockContext(); // TODO create separate CmsContext?
+    const fileInputRef = useRef<HTMLInputElement>(null);
+
+    const maxFileSizeInMegabytes = cmsBlockContext.damConfig.maxFileSize;
+    const maxFileSizeInBytes = maxFileSizeInMegabytes * 1024 * 1024;
+    const cancelUpload = useRef<CancelTokenSource>(axios.CancelToken.source());
+    const errorDialog = useErrorDialog();
+    const [replaceLoading, setReplaceLoading] = useState(false);
+
+    const { getInputProps } = useDropzone({
+        maxSize: maxFileSizeInBytes,
+        multiple: false,
+        accept: convertMimetypesToDropzoneAccept([file.mimetype]),
+        onDrop: async (acceptedFiles: File[], fileRejections: FileRejection[]) => {
+            if (fileRejections.length > 0) {
+                errorDialog?.showError({
+                    userMessage: (
+                        <FormattedMessage
+                            id="comet.dam.file.replace.fileRejection"
+                            defaultMessage="The selected file could not be uploaded because it doesn't meet the required criteria. Please choose a valid file to replace the existing one."
+                        />
+                    ),
+                    error: fileRejections.toString(),
+                });
+            }
+
+            try {
+                setReplaceLoading(true);
+                const response = await replaceById({
+                    apiClient: cmsBlockContext.damConfig.apiClient,
+                    data: { file: acceptedFiles[0], fileId: file.id },
+                    cancelToken: cancelUpload.current.token,
+                });
+
+                if (response.status === 201 && response.data) {
+                    const fileUrl = (response.data as { fileUrl?: string })?.fileUrl;
+                    if (fileUrl) {
+                        apolloClient.cache.modify({
+                            id: `DamFile:${file.id}`,
+                            fields: {
+                                fileUrl: () => fileUrl,
+                            },
+                        });
```

Comment:
Is this the only change that's necessary? What about the crop area? Wouldn't it be better to evict the cache entry?

Reply:
You are right, fileUrl isn't enough dba0f4e22562782bbc4c03ca2a93fe3a4d184bb0